src/instructlab/model/backends/backends.py

```diff
@@ -304,6 +309,100 @@ def shutdown_process(process: subprocess.Popen, timeout: int) -> None:
         )
         process.kill()
 
+    # Various facilities of InstructLab rely on multiple successive start/stop
+    # cycles. Since vLLM relies on stable VRAM for estimating capacity, residual
```

Comment:
This seems reversed. If we need "stable VRAM" on vLLM startup, then shouldn't the wait happen when it starts? If this is the last vLLM run, the wait below will be for nothing, no?

Reply:
Sure we could do it either way, we have a similar problem were on start the waiting may not be necessary. The best long term fix would be to do a patch to vlllm to alter its mechanism. The only reason I leaned toward doing this on stop was just that the start sequence is incredibly long as it is.

Reply:
I would vote for not changing it given the amount of testing that has gone into this approach.  In either case we would probably want to start passing flags indicating whether the check was necessary.  That should eliminate most of the concern.  And hopefully we can get a fix in vllm to avoid this in the long run.

Reply:
I see how this is catch-22, one way or another it slows things down. I don't insist on this to change.

Reply:
@n1hility is this reported somewhere in vllm? Would be nice to add a link to the report in comments so that readers can follow up if / when this is fixed.

Reply:
sure will do